src/main/java/nl/rkslot/nlapi_test_suite/NamelessApiTest.java

```diff
@@ -1,42 +1,126 @@
+                        m.invoke(testStage, api);
+                    } else {
+                        m.invoke(testStage);
+                    }
+                } catch (IllegalAccessException | InvocationTargetException e) {
+                    e.printStackTrace();
+                }
+            });
+
+            List<Method> tests = Arrays.stream(testStage.getClass().getDeclaredMethods()).filter(method -> {
+                return method.isAnnotationPresent(Test.class)
+                        && method.getParameterCount() == 1
+                        && method.getName().startsWith("test_");
+            }).collect(Collectors.toList());
+
+            if (tests.isEmpty()) {
+                System.out.println("⚠️  No tests found in " + testStage.getClass().getSimpleName());
+            } else {
+                tests.forEach(method -> {
+                    final long startTest = Calendar.getInstance().getTimeInMillis();
+                    testCount.getAndIncrement();
+
+                    final @NotNull String testName = getTestName(method.getName());
+
+                    System.out.println("⏳ Starting test: " + testName + "");
+
+                    boolean pass = true;
+
+                    try {
+                        method.invoke(testStage, api);
+                    } catch (IllegalAccessException e) {
+                        e.printStackTrace();
+                    } catch (InvocationTargetException e) {
+                        if (e.getCause() instanceof AssertionError || e.getCause() instanceof ApiError) {
+                            // print the assertion error or api error directly, don't waste screen space on the whole stacktrace
+                            e.getCause().printStackTrace();
+                        } else {
+                            e.printStackTrace();
+                        }
+                        pass = false;
+                    }
+
+                    final long finishedTest = Calendar.getInstance().getTimeInMillis();
+                    final long taken = finishedTest - startTest;
+
+                    if (pass) {
+                        System.out.println("✅ Test passed: " + testName + " (" + taken + "ms)");
+                        testsPassed.getAndIncrement();
+                    } else {
+                        System.out.println("❌ Test failed: " + testName + " (" + taken + "ms)");
+                    }
+                });
+            }
+
+            System.out.println();
+        }
+
+        long ended = Calendar.getInstance().getTimeInMillis();
+
+        System.out.println("---------------------- All tests completed ----------------------");
+        System.out.println("➡️  " + testsPassed + "/" + testCount + " tests passed");
+        System.out.println("➡️  Made " + TestStage.getAssertions() + " assertions");
+        System.out.println("➡️  Took " + (ended - started) + "ms");
+    }
```

Comment:
Now that you catch any exceptions, the program should manually call `System.exit(1)` if any test fails